scisim/api.py

```diff
@@ -0,0 +1,115 @@
+from scisim import app, db, clean_text
+from flask import request
+from models import *
+from scisim.helpers import *
+from sqlalchemy import and_
+from datetime import datetime
+from json import dumps
+
+@app.route('/api/login')
+def login():
+    username = request.form['username']
+    sim_id = request.form['sim_id']
+    password = request.form['password']
+
+    if len(username) < 1:
+        return error_message('The username must have more than 1 character')
+
+    sim = Simulation.query.filter(Simulation.id == sim_id).first()
+    if sim.password != password:
+        return error_message('The simulation password is not correct')
+
+    user = User(name=username, sim_id=sim_id)
+    db.add(user)
+    db.commit()
+
+    return dumps(unpack_model(user))
+
+
+@app.route('/api/logout')
+def logout():
+    # maybe we could create a 'logged in' table that will store logged in users. Put an expiration and remove logged in users with a cron job
+    pass
+
+@app.route('/api/simulations', methods=['GET'])
+def simulations():
+    return to_json(Simulation.query.all())
+
+@app.route('/api/pages', methods=['POST'])
+def pages():
+    check = check_for_params(['sim_id'], request)
+    if type(check) is type(str()):
```

Comment:
This seems awkward. What if check_for_params returned None for success, and a string error message otherwise. Then the code becomes:
error = check_for_params(...)
if error:
    return error_message(error)